### DISCUSSION TITLE
Feature request: Search by multiple prefixes

### DISCUSSION THREAD

**Discussion by der3k** [meilisearch/product#640] [Feedback & Feature Proposal]
Currently only the last word in the multi-word query is used for prefix search, rest of the query words are exact match searches.
I would like to use every word in a multi-word query for prefix search.

For example, when searching for "Apple watch" I would like the query to be "app wat" and still get the result.

As far as I know its not possible with the current version of meilisearch, or am I missing something?

Ideally it would be great to specify which word in multi-word search query should prefix match and which should do exact match.
In this way it would be possible to prefix match on text and exact match numbers in one query. Great for address searching.

For example query "ba st 221 lon" should return 'Baker Street 221 in London.

---

Edit from @curquiza, with related examples
- https://github.com/meilisearch/meilisearch/issues/4093#issuecomment-1755322763
- https://github.com/meilisearch/product/discussions/97#discussioncomment-2014212

**Comment by macraig** [comment]
Hi @der3k , thank you for taking the time to open a discussion and apologies for the late reply 🙏 

> As far as I know its not possible with the current version of meilisearch, or am I missing something?

This is correct, we currently only consider the last word in your query as incomplete and therefore that's the only work that will have a prefix search.

> Ideally it would be great to specify which word in multi-word search query should prefix match and which should do exact match.
In this way it would be possible to prefix match on text and exact match numbers in one query. Great for address searching.

I agree! We currently have an open card related to this Under Consideration on our [ public roadmap](https://roadmap.meilisearch.com/c/61-policy-for-prefix-search) in case you want to upvote the feature there.
